core/src/main/java/io/questdb/griffin/engine/table/ShowCreateTableRecordCursorFactory.java

```diff
@@ -0,0 +1,273 @@
+                        sink.put(" PARTITION BY ").put(table.getPartitionByName());
+                    }
+
+                    // (BYPASS) WAL
+                    if (!table.getWalEnabled()) {
+                        sink.put(" BYPASS");
+                    }
+                    sink.put(" WAL");
+                }
+
+                final CairoConfiguration config = executionContext.getCairoEngine().getConfiguration();
+                final boolean withMaxUncommittedRows = table.getMaxUncommittedRows() != config.getMaxUncommittedRows();
+                final boolean withO3MaxLag = table.getO3MaxLag() != config.getO3MaxLag();
+                final boolean withRequired = withMaxUncommittedRows || withO3MaxLag;
+
+                // WITH maxUncommittedRows=123, o3MaxLag=456s
+                if (withRequired) {
+                    sink.put('\n').put("WITH ");
+                    if (withMaxUncommittedRows) {
+                        sink.put("maxUncommittedRows=").put(table.getMaxUncommittedRows());
+                    }
+                    if (withO3MaxLag) {
+                        if (withMaxUncommittedRows) {
+                            sink.put(", ");
+                        }
+                        sink.put("o3MaxLag=").put(table.getO3MaxLag()).put("us");
+                    }
+                }
+
+                // IN VOLUME OTHER_VOLUME
+                if (table.getIsSoftLink()) {
+                    if (withRequired) {
+                        sink.put(',');
+                    }
+                    sink.put(" IN VOLUME ");
+
+                    FilesFacade ff = config.getFilesFacade();
+                    try (Path softLinkPath = new Path().of(config.getRoot()).concat(table.getDirectoryName())) {
+                        try (Path otherVolumePath = new Path()) {
+                            ff.readLink(softLinkPath, otherVolumePath);
+                            otherVolumePath.trimTo(otherVolumePath.size()
+                                    - table.getDirectoryName().length()  // look for directory
+                                    - 1 // get rid of trailing slash
+                            );
+                            CharSequence alias = config.getVolumeDefinitions().resolvePath(otherVolumePath.asAsciiCharSequence());
+
+                            if (alias == null) {
+                                throw CairoException.nonCritical().put("could not find volume alias for table [table=").put(tableToken).put(']');
+                            } else {
+                                sink.put(alias);
+                            }
+                        }
+                    }
+                }
+
+                // DEDUP UPSERT(key1, key2)
+                if (table.getIsDedup()) {
+                    sink.put('\n');
+                    sink.put("DEDUP UPSERT KEYS(");
+                    for (int i = 0, n = (int) table.getColumnCount(); i < n; i++) {
```

Comment:
why getColumnCount() returns 64-bit value? How many columns do we expect users to have?